sections/Design.tex

```diff
@@ -34,8 +34,76 @@ \subsection{Premade functions (rename)}
 
-\subsection{While-loop}
+The conditionals in many programming languages are: if, ternary operator (?:) and switch.
+The combination of a conditional and a jump, often seen as a \texttt{goto} can produce an iteration. 
+The goto statement is not in our language, it is considered harmful by many, and in all but a few cases useless, if other conditionals are in the programming language. \citep{DijkstraGoto}
+The switch statement is not included in our language. \todo{Skal vi have switch med?}
 
-\subsection{If \& If else}
+\subsection{While-loop}
+The while-loop is a very common language construct, it iterates once or more times \textit{while} a given boolean condition is true. 
+In our language blocks are delimited by curly brackets (``\texttt{\{}'' and ``\texttt{\}}''), and so is the body of the while-loop. 
+Unlike languages like C, C\# and Java etc. in our language it is not allowed to omit the curly brackets if one wishes to only to have a single statement in the while-loop, this is a syntax error. 
+This is to increase the consistency of the code thus making it more readable. 
+In \myref{lst:whileExample} is an example of a while-loop in our language. 
+An variation of the while-loop called the do..while-loop, in which a single execution a done before checking the boolean condition is not included in our language. 
+This is to reduce the amount of built-in control structures. 
+
+\begin{figure}[h]
+\begin{lstlisting}[caption=An example of a while-loop, label=lst:whileExample]
+while ( boolean_condition )
+{
+    example_statement();
+}
+\end{lstlisting}
+\end{figure}
+\subsection{For-loop}
+A for-loop is a loop like the while-loop, with additional features. 
+The traditional for-loop has three fields, the initialization, the condition and the increment/decrement. 
+Any of these can be omitted, if the conditional is omitted then the loop is an infinite loop, but it can still be broken by the \texttt{break} statement.
+Like for while-loops, curly brackets around the body of for-loops are enforced. 
+Any traditional for-loop can be translated into a while-loop, by putting the initialization before the while-loop and the increment/decrement in the end of the body. 
+We have the tradition for-loop in our language, an example of it is shown in \myref{lst:whileExample}. 
+For-loops are often used to iterate over a collection, and it is easier to see what the for-loop does than a while-loop, as the three fields are in one place. 
+This can be further simplified by having a foreach-loop, which iterate over each element in a collection.
+However this is not included in our language, in order to reduce the scope of the language and simplify its development. \todo{Skal vi have foreach?}
+\begin{figure}[h]
+\begin{lstlisting}[caption=An example of a for-loop, label=lst:forExample]
+for ( initialization ; boolean_condition ; increment/decrement )

```

Comment:
Man kunne evt. vise andre eksempler på syntaksen der kunne bruges.

for(1 to 32){}
eller lignende, og hvorfor vi gerne vil bruge den vi har valgt her. måske henvend til filosofien.